### PR TITLE
Fix qubits indexing in Statevector._evolve_instruction

### LINKED ISSUE #5282: Statevector.from_instruction fails for custom controlled gates
### Information

- **Qiskit Terra version**: latest
- **Python version**:
- **Operating system**:

### What is the current behavior?
```python
import numpy as np
from qiskit import QuantumCircuit
from qiskit.quantum_info import Statevector

## Create 7mod15 gate
N = 15
m = int(np.ceil(np.log2(N)))

U_qc = QuantumCircuit(m)
U_qc.x(range(m))
U_qc.swap(1, 2)
U_qc.swap(2, 3)
U_qc.swap(0, 3)

U = U_qc.to_gate()
U.name ='{}Mod{}'.format(7, N)
U_cntrl = U.control()

qc = QuantumCircuit(5)
qc.append(U_cntrl, range(5))

Statevector.from_instruction(qc)
```
gives

Traceback (most recent call last):
  File "<ipython-input-127-b464b6ab1295>", line 1, in <module>
    Statevector.from_instruction(qc)
  File "/opt/miniconda3/envs/qiskit/lib/python3.7/site-packages/qiskit/quantum_info/states/statevector.py", line 529, in from_instruction
    return Statevector._evolve_instruction(vec, instruction)
  File "/opt/miniconda3/envs/qiskit/lib/python3.7/site-packages/qiskit/quantum_info/states/statevector.py", line 690, in _evolve_instruction
    Statevector._evolve_instruction(statevec, instr, qargs=new_qargs)
  File "/opt/miniconda3/envs/qiskit/lib/python3.7/site-packages/qiskit/quantum_info/states/statevector.py", line 690, in _evolve_instruction
    Statevector._evolve_instruction(statevec, instr, qargs=new_qargs)
  File "/opt/miniconda3/envs/qiskit/lib/python3.7/site-packages/qiskit/quantum_info/states/statevector.py", line 661, in _evolve_instruction
    return Statevector._evolve_operator(statevec, Operator(mat), qargs=qargs)
  File "/opt/miniconda3/envs/qiskit/lib/python3.7/site-packages/qiskit/quantum_info/states/statevector.py", line 642, in _evolve_operator
    np.reshape(statevec.data, pre_tensor_shape), axes), contract_shape)
  File "<__array_function__ internals>", line 6, in transpose
  File "/opt/miniconda3/envs/qiskit/lib/python3.7/site-packages/numpy/core/fromnumeric.py", line 653, in transpose
    return _wrapfunc(a, 'transpose', axes)
  File "/opt/miniconda3/envs/qiskit/lib/python3.7/site-packages/numpy/core/fromnumeric.py", line 58, in _wrapfunc
    return bound(*args, **kwds)
ValueError: axes don't match array

### PR DESCRIPTION
**Pull request by georgios-ts**
### Summary
Resolves #5282.

### Details and comments
#5282 was failing due to bad indexing in `_evolve_instruction` in case the instruction contains multiple registers.

### CODE REVIEW

File: qiskit/quantum_info/states/statevector.py [hunk 1]
```diff
@@ -677,15 +677,16 @@ def _evolve_instruction(statevec, obj, qargs=None):
                 obj.name, type(obj.definition)))
         if obj.definition.global_phase:
             statevec._data *= np.exp(1j * float(obj.definition.global_phase))
+        qubits = obj.definition.qubits
```
Comment (kdk) [inline qiskit/quantum_info/states/statevector.py:680]: `qubits.index(tup)` will repeatedly search through `qubits` to find `tup`, which will likely be slow for large instructions. Can this be a map like
```
qubits = {qubit: i for i, qubit in enumerate(obj.definition.qubits)}
```
? Then `new_qargs` could be something like `[qubits[tup] for tup in qregs]`.

**Review by kdk — COMMENTED** [review]
Thanks @georgios-ts , this looks good. One comment below.

**Comment by kdk** [comment]
Also, can you add a release note?

File: qiskit/quantum_info/states/statevector.py (hunk 1 above)
Comment (georgios-ts) [reply qiskit/quantum_info/states/statevector.py]: Yes, good point. Now i use a map. Also, release note added.

**Review by 1ucian0 — APPROVED** [review]

**Pull request merged**